app/views/additional_payments/claims/entire_term_contract.html.erb

```diff
@@ -1,42 +1,34 @@
-<% content_for(:page_title, page_title(t("additional_payments.questions.has_entire_term_contract"), journey: current_journey_routing_name, show_error: current_claim.errors.any?)) %>
-<% path_for_form = current_claim.persisted? ? claim_path(current_journey_routing_name) : claims_path(current_journey_routing_name) %>
+<% content_for(:page_title, page_title(t("additional_payments.forms.entire_term_contract.questions.has_entire_term_contract"), journey: current_journey_routing_name, show_error: @form.errors.any?)) %>
 
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
-    <%= render("shared/error_summary", instance: current_claim, errored_field_id_overrides: { "eligibility.has_entire_term_contract": "claim_eligibility_attributes_has_entire_term_contract_true" }) if current_claim.errors.any? %>
-    <%= form_for current_claim, url: path_for_form do |form| %>
-      <%= form_group_tag current_claim do %>
-        <%= form.fields_for :eligibility, include_id: false do |fields| %>
+    <%= render("shared/error_summary", instance: @form, errored_field_id_overrides: { "eligibility.has_entire_term_contract": "claim_eligibility_attributes_has_entire_term_contract_true" }) if @form.errors.any? %>
```

Comment:
I think the overrides need to change to:

```
errored_field_id_overrides: { "has_entire_term_contract": "claim_has_entire_term_contract_true"}
```

Technically, the `has_entire_term_contract` is indeed an eligibility attribute, but the `@form` object has no knowledge of the `eligibility` entity (other than saving on it). We'll definitely remove this separation claim vs eligibility at some point soon, see https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2666#discussion_r1553981319

Reply:
ooo good catch, thank you!